cmd/gaia/app/genesis.go

```diff
@@ -90,19 +90,20 @@ func GaiaAppInit() server.AppInit {
 
 // Create the core parameters for genesis initialization for gaia
 // note that the pubkey input is this machines pubkey
-func GaiaAppGenState(cdc *codec.Codec, appGenTxs []json.RawMessage) (genesisState GenesisState, err error) {
-	if len(appGenTxs) == 0 {
-		err = errors.New("must provide at least genesis transaction")
+func GaiaAppGenState(cdc *codec.Codec, genDoc tmtypes.GenesisDoc, appGenTxs []json.RawMessage) (
+	genesisState GenesisState, err error) {
+
+	if err = cdc.UnmarshalJSON(genDoc.AppState, &genesisState); err != nil {
 		return
 	}
 
-	// start with the default staking genesis state
-	stakeData := stake.DefaultGenesisState()
-	slashingData := slashing.DefaultGenesisState()
-
-	// get genesis flag account information
-	genaccs := make([]GenesisAccount, len(appGenTxs))
+	// if there are no gen txs to be processed, return the default empty state
+	if len(appGenTxs) == 0 {
+		err = errors.New("there must be at least one genesis tx")
+		return
```

Comment:
Let's use explicit return variables _at_ each return statement here (aka `return genesisState, err`) 
I know not all of the code is currently structured this way but let's update it to be to this way for the functions which are modified in this PR